File: euler_074.py
from time import time
from math import factorial

def sum_dig(n):
    return sum([f[int(d)] for d in str(n)])

start = time()

f = {n:factorial(n) for n in range(10)}

s = [sum_dig(n) for n in range(10**6)]

s += [sum_dig(n) for n in range(10**6, max(s) + 1)]

remaining = [True] * (10**6 + 1)

sixties = 0

for i in range(len(remaining)):
    r = remaining[i]
    if r:
        chain = {i}
        n = s[i]
        while n not in chain:
            chain.add(n)
            n = s[n]
        if len(chain) == 60:
            sixties += 1
        for e in chain:
            if e < 10**6:
                remaining[e] = False

print("Number of 60-length chains: " + str(sixties))


stop = time()
print("Time: " + str(stop - start))

File: euler_074.js
const start = Date.now();

let factorial = [1,1];
for (let n = 2; n < 10; n++) {
    factorial[n] = n * factorial[n - 1];
}

const factorialSum = (num) => {
    return num.toString().split("").map(x => Number(x)).map(x => factorial[x]).reduce((p,c) => p + c, 0);
}

let checked = Array(1000000);
checked.fill(false);
checked[0] = true;
checked[1] = true;

let longChains = 0;


for (let n = 2; n < 1000000; n++) {
  if (!checked[n]) {
    let chain = [n];
    let fs = factorialSum(n);

    while (chain.indexOf(fs) === -1) {
      chain.push(fs);
      fs = factorialSum(fs);
    }

    if (chain.length < 60) {
        new Set(chain).forEach((x) => {
            checked[x] = true;
        });
    }

    if (chain.length === 60) {
        longChains++;
        new Set(chain).forEach((x) => {
            checked[x] = true;
        });
    }
  }
}


const end = Date.now();

console.log("Execute time: ", end - start, " The numerator is: ", longChains);